packages/abi-coder/src/encoding/strategies/getCoderV1.ts

```diff
@@ -114,7 +113,15 @@ export const getCoder: GetCoderFn = (
   }
 
   if (resolvedAbiType.type === VEC_CODER_TYPE) {
-    const arg = findOrThrow(components, (c) => c.name === 'buf').originalTypeArguments?.[0];
+    const bufferComponent = components.find((c) => c.name === 'buf');
+    if (!bufferComponent) {
+      throw new FuelError(
+        ErrorCode.INVALID_COMPONENT,
+        `The provided Vec type is missing the 'buf' component.`
+      );
+    }
+
+    const arg = bufferComponent.originalTypeArguments?.[0];
```

Comment:
May I'm missing something but aren't these lines just a duplicate of [CoderV0](https://github.com/FuelLabs/fuels-ts/pull/2000/files#diff-0ee699e8c8af0ac5e0e4ed4059dc3a58071ec13a4a31934c92f29918a8a24ebcR113-R121) perhaps this should be shared function.